cpp/src/utilities/traits.cpp

```diff
@@ -195,6 +195,7 @@ struct is_numeric_not_bool_impl {
 
 bool is_numeric_not_bool(data_type type)
 {
+  if (type.id() == type_id::EMPTY) return false;
```

Comment:
I don't think we want to support this. It circumvents the type-dispatcher logic which is the intended result.
The `EMPTY` is not a real type and I feel `false` is the incorrect response and the exception is the desired result.
I would be worried that calling logic is expecting `false` only for valid (non-EMPTY) types.
The caller can catch this exception and handle it appropriately.

Reply:
That makes sense. It's probably worth raising a more friendly python error in pylibcudf because currently the error you get is.
```
terminate called after throwing an instance of 'cudf::logic_error'
  what():  CUDF failure at:/cpp/include/cudf/utilities/type_dispatcher.hpp:546: Invalid type_id.
Aborted (core dumped)
```
And I think it kills your interactive python session.

Reply:
Could `type_dispatcher` have a `case` for `type_id::EMPTY` so an `is_empty_impl` functor can be added?

Reply:
I would be against supporting an `is_empty_type()` function and reworking the type-dispatcher for this.
Just check the `type_id::EMPTY` in your calling code if you need to.
We would ultimately like to get rid of this type from libcudf but it is required for some Arrow interop right now.
All libcudf functions are expected to throw if this type is used.

Reply:
Makes sense. Thanks for the context

Reply:
@Matt711 @mroeschke Can one of you post up a MWE of where the std::terminate mentioned above happens when using pylibcudf? 

Practically speaking I'm not sure if we'll ever be able to fully get rid of EMPTY without losing lossless round-tripping with arrow (which does have very slightly more robust handling of the null type than we do of empty), but we can keep it as slim as possible.

Reply:
```
import pylibcudf as plc
plc.traits.is_numeric_not_bool(plc.DataType(plc.TypeId.EMPTY))
```

Reply:
I just saw that there is an open issue with this, sorry about that.

Reply:
I reopened the issue with a suggestion for what we should do.

Reply:
Thanks